assertk/src/commonMain/kotlin/assertk/failure.kt

```diff
@@ -139,17 +140,29 @@ fun fail(error: AssertionError): Nothing {
 
 internal val NONE: Any = Any()
 
+// TODO Delete this before 1.0.
+@Deprecated("For binary compatibility", level = HIDDEN)
+fun fail(message: String, expected: Any? = NONE, actual: Any? = NONE): Nothing {
```

Comment:
I really haven't been bothering with binary compat before 1.0 (as opposed to source compat) with the assumption being that assertk is likely to be a leaf dependency instead of a transitive one. Would be happy to hear your thoughts though.

Reply:
It seemed like extremely low effort to retain so I included it. I could see a published library of assertions existing which might have used the function. That being said, if you don't want to set a precedent or anything I'm not tied to it.

Reply:
yeah it doesn't hurt anything, not sure I want to make binary compat a requirement for updates but I won't block merging this in on that.